client-base/src/main/kotlin/app/cash/backfila/client/spi/BackfilaParameters.kt

```diff
@@ -33,8 +36,20 @@ class BackfilaParametersOperator<T : Any>(
   /** Constructor parameters used as defaults when missing to create a new T. */
   private val constructor: KFunction<T> = fetchConstructor(parametersClass)
 
-  fun constructBackfillConfig(
+  // Creating a new config would be too heavy-handed in my opinion, so I just put in what should be
+  // an obvious default.
+  fun constructBackfillConfig(request: PrepareBackfillRequest): BackfillConfig<T> =
+      constructBackfillConfig(request.parameters, "N/A - PrepareBackfill", request.dry_run)
```

Comment:
maybe just null for partitionName instead?

Reply:
I thought about that but figured that might cause more errors? I figured this might be better and more obvious than a nullpointer?

If you and Keefer think I'm crazy I can make it null sure.

Reply:
well if its marked nullable, callers can just check. I guess I'm not sure what the use cases would really be

Reply:
ok ended up just creating a PrepareBackfillConfig class instead. It will cause some short term pain but probably the better model tbh.